src/event-stream/reader.ts

```diff
@@ -74,53 +112,30 @@ export function parseMessageTransactions(msg: CoreNodeBlockMessage): CoreNodeMes
     const coreTx = msg.transactions[i];
     try {
       const txBuffer = Buffer.from(coreTx.raw_tx.substring(2), 'hex');
-      const bufferReader = BufferReader.fromBuffer(txBuffer);
-      let txSender: string;
-      let sponsorAddress = undefined;
       let rawTx: Transaction;
-      try {
+      let txSender: string;
+      let sponsorAddress: string | undefined;
+      if (coreTx.raw_tx === '0x00') {
+        const event = msg.events.find(event => event.txid === coreTx.txid);
```

Comment:
This was previously doing `event = msg.events[i]` where `i` was the tx index rather than event index.